File: skeletons/backend/express-skeleton/src/handlers/health.ts
import { Request, Response } from "express";

interface HealthResponse {
  status: string;
  message: string;
}

/**
 * Returns the current health status of the application.
 */
export function healthCheck(_req: Request, res: Response): void {
  const response: HealthResponse = {
    status: "healthy",
    message: "Express.js REST API application is running",
  };

  res.status(200).json(response);
}

File: skeletons/go-skeleton/internal/handlers/health.go
package handlers

import (
	"encoding/json"
	"net/http"
)

// HealthResponse represents the health check response body.
type HealthResponse struct {
	Status  string `json:"status"`
	Message string `json:"message"`
}

// HealthCheck returns the current health status of the application.
func HealthCheck(w http.ResponseWriter, r *http.Request) {
	resp := HealthResponse{
		Status:  "healthy",
		Message: "Go REST API application is running",
	}

	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(http.StatusOK)
	json.NewEncoder(w).Encode(resp)
}
